closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -1670,10 +1670,28 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_RSADP
                 if (-1 != msg_len2) {
 
                     int i;
-                    for (i = 0; i < verify; i++) {
-                        if (kNative[i] != k2[i]) {
+                    /*
+                     * For certain key sizes, the decrypted message retrieved from the RSA_public_decrypt
+                     * includes a 1 byte padding at the beginning of the message. In these cases, this
+                     * padding must be ignored when comparing to the original message.
```

Comment:
Please correct this comment. We're not ignoring that padding; instead, we expect it to be a zero byte.

Reply:
Right, fixed it.